hazelcast/src/main/java/com/hazelcast/jet/core/processor/RemoteMapSourceParams.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core.processor;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.query.Predicate;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Parameters to use a remote map as source
+ *
+ * @param <T> specifies emitted type
+ * @param <K> specifies key type
+ * @param <V> specifies value type
+ */
+public class RemoteMapSourceParams<T, K, V> {
```

Comment:
This will protect the internal layers from change when a new parameter is added.  Passing X parameters through the stack is still a problem. 
For example when we code the source builder, it will populate a parameter object and pass to internal layer. Then we can easily add more functionality